src/main/java/seedu/ddd/model/contact/common/ContactPhonePredicate.java

```diff
@@ -0,0 +1,17 @@
+package seedu.ddd.model.contact.common;
+
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@Code Contact} 's {@code Phone} matches the given phone number.
+ */
+public class ContactPhonePredicate implements Predicate<Contact> {
+    private final Phone phoneNumber;
+    public ContactPhonePredicate(Phone phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+    @Override
+    public boolean test(Contact contact) {
+        return contact.getPhone().value.contains(this.phoneNumber.value);
+    }
```

Comment:
Same comment as `ContactEmailPredicate`. But maybe it might more sense to search for `phoneNumber` by prefix instead of just checking if it contains?

Reply:
was thinking we if the user forgets the user's number and only knows partially, this would still work